superset/viz.py

```diff
@@ -865,6 +865,12 @@ def query_obj(self) -> QueryObjectDict:
             raise QueryObjectValidationError(_("Please choose at least one metric"))
         if set(groupby) & set(columns):
             raise QueryObjectValidationError(_("Group By' and 'Columns' can't overlap"))
+        sort_by = self.form_data.get("timeseries_limit_metric")
+        if sort_by:
+            sort_by_label = utils.get_metric_name(sort_by)
+            if sort_by_label not in d["metrics"]:
+                d["metrics"].append(sort_by)
+            d["orderby"] = [(sort_by, not self.form_data.get("order_desc", True))]
```

Comment:
it is may be more readable
```suggestion
d["orderby"] = [(sort_by, bool(self.form_data.get("order_desc"))]
```

Reply:
This is incorrect. The second parameter is the `orderby` tuple is "is_ascending", which is the opposite meaning of `order_desc`. However, we also want the default to be order in descending in case `order_desc` is not set.

Reply:
thanks for the explanation, it's my fault